### PR TITLE
Indexing fixes

### LINKED ISSUE #1425: Fill the output of af_create_indexers with af_span by default
I wrote the following test program in C (skipped out the function and header files): 
```c
    af_index_t* indexArr =  (af_index_t*)(malloc(4 * sizeof(af_index_t)));
    af_err err = af_create_indexers(&indexArr);
    printf("Index_t size:  %lu \n", sizeof(af_index_t));
    printf("Error code:  %d \n", err);

    //Create array
    af_array a;
    unsigned ndims = 2;
    dim_t dim[] = {10,10};
    af_randu(&a, ndims, dim, f32);
    af_print_array(a);

    //Create index array
    af_array idx;
    unsigned n = 1;
    dim_t d[] = {5};
    af_range(&idx, n, d, 0, s64);
    af_print_array(idx);

    //create array indexer
    af_err err2 = af_set_array_indexer(indexArr, idx, 1);
    printf("Error from set array indexer: %d \n", err2);

    //index with indexArr
    af_array out;
    af_index_gen(&out, a, 1, indexArr);  //segfaults
    af_print_array(out);
    printf("Yay this works! \n");

    //create another array indexer 
    af_err err3 = af_set_array_indexer(indexArr, idx, 0);
    printf("Error from second array indexer: %d \n", err3);

    //index with indexArr
    af_array out2;
    af_index_gen(&out2, a, 2, indexArr);
    af_print_array(out2);
```
When I set talk about indexing in the second dimension, my index function segfaults at that particular line with the comment. This is because the `create_indexers` creates garbage objects by default. 

After a discussion with @9prady9 and @pavanky , it seems that by default, `af_create_indexers` should return `span,span,span,span` so users can set any dimension directly.

### PR DESCRIPTION
**Pull request by 9prady9**
Fixes #1425 

[skip ci]

### CODE REVIEW

**Comment by 9prady9** [comment]
@pavanky I added code example near the `af_create_indexers` function and included all the these helper functions under `index` category.

**Comment by pavanky** [comment]
@9prady9 can you make the example from the documentation into a test and include it as a code snippet from the documentation ?

**Comment by 9prady9** [comment]
@pavanky Please check now.

File: test/index.cpp [hunk 1]
```diff
@@ -495,6 +495,49 @@ TYPED_TEST(Indexing, 3D_to_1D)
     DimCheckND<TypeParam>(this->continuous3d_to_1d, TEST_DIR"/index/Continuous3Dto1D.test", 3);
 }
 
+
+TEST(Index, Docs_Util_C_API)
+{
+    //![ex_index_util_0]
+    af_index_t* indexers = 0;
+    af_err err = af_create_indexers(&indexers); // Memory is allocated on heap by the callee
```
Comment (pavanky) [inline test/index.cpp]: Need not get err back here

File: test/index.cpp [hunk 2]
```diff
@@ -495,6 +495,49 @@ TYPED_TEST(Indexing, 3D_to_1D)
...
+    af_array a;
+    unsigned ndims = 2;
+    dim_t dim[] = {10,10};
+    af_randu(&a, ndims, dim, f32);
+
+    //Create index array
+    af_array idx;
+    unsigned n = 1;
+    dim_t d[] = {5};
+    af_range(&idx, n, d, 0, s64);
```
Comment (pavanky) [inline test/index.cpp]: Make this s32

File: test/index.cpp [hunk 3]
```diff
@@ -495,6 +495,49 @@ TYPED_TEST(Indexing, 3D_to_1D)
...
+
+    af_print_array(a);
+    af_print_array(idx);
+
+    //create array indexer
+    err = af_set_array_indexer(indexers, idx, 1);
+    if (err != AF_SUCCESS) {
+        printf("Error from set array indexer: %d \n", err);
+        exit(1);
```
Comment (pavanky) [inline test/index.cpp]: need not check error here either.

File: test/index.cpp [hunk 4]
```diff
@@ -495,6 +495,49 @@ TYPED_TEST(Indexing, 3D_to_1D)
...
+    }
+
+    //index with indexers
+    af_array out;
+    af_index_gen(&out, a, 2, indexers); // number of indexers should be two since
```
Comment (pavanky) [inline test/index.cpp]: Check for error here.

File: test/index.cpp [hunk 5]
```diff
@@ -495,6 +495,49 @@ TYPED_TEST(Indexing, 3D_to_1D)
...
+                                        // we have set only second af_index_t
+    af_print_array(out);
```
Comment (pavanky) [inline test/index.cpp]: May be add another couple of lines to showcase `af_set_seq_indexer` and `af_set_seq_param_indexer` ?

**Comment by pavanky** [comment]
Merged it into https://github.com/arrayfire/arrayfire/pull/1427, closing this.

**Pull request closed without merging**